File: go/rna-transcription/rna_transcription.go
package strand

import "strings"

func ToRNA(dna string) string {
	dnaRnaMap := map[string]string {
		"G": "C",
		"C": "G",
		"T": "A",
		"A": "U",
	}

	rna := ""

	for _, el := range dna {
		val, ok := dnaRnaMap[string(el)]

		if ok {
			rna += val
		}
	}

	// 	return toRNA.Replace(s)
	return rna
}

var toRNA = strings.NewReplacer(
	"G", "C",
	"C", "G",
	"T", "A",
	"A", "U",
)

File: javascript/rna-transcription/rna-transcription.js
export const toRna = (dna) => {
  return dna.split('').map((letter) => dnaRnaMap[letter]).join('');
};

const dnaRnaMap = {
  'G': 'C',
  'C': 'G',
  'T': 'A',
  'A': 'U'
};
